src/blockfilter.cpp

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <blockfilter.h>
+#include <hash.h>
+#include <primitives/transaction.h>
+#include <script/script.h>
+#include <streams.h>
+
+/// SerType used to serialize parameters in GCS filter encoding.
+constexpr int GCS_SER_TYPE = SER_NETWORK;
```

Comment:
I'm curious what triggered this request. I may be reading the wrong places, but e.g. https://stackoverflow.com/questions/45987571/difference-between-constexpr-and-static-constexpr-global-variable indicates static does nothing at all here.

Reply:
My thought was to make these explicitly translation-unit local, to communicate that to the reader and to enforce that from the compiler. Here's what cppreference says about what gets internal linkage:

> internal linkage. The name can be referred to from all scopes in the current translation unit.
Any of the following names declared at namespace scope have internal linkage:
* variables, functions, or function templates declared static;
* non-volatile non-inline (since C++17) const-qualified variables (including constexpr) that aren't declared extern and aren't previously declared to have external linkage;
* data members of anonymous unions.

https://en.cppreference.com/w/cpp/language/storage_duration#Linkage

Worth noting that non-static constexpr variables can be declared extern in other contexts and rendered external linkage, while static variables can not be.

Unnamed namespaces are another way to enforce this:
> Even though names in an unnamed namespace may be declared with external linkage, they are never accessible from other translation units because their namespace name is unique.
https://en.cppreference.com/w/cpp/language/namespace#Unnamed_namespaces

Reply:
@kallewoof Interesting. I looked it up too and came across this: https://stackoverflow.com/a/13867690.

Reply:
@jimpo Yeah I read that one too, but the last paragraph was what made me unsure in the end whether it was good or bad

> However, there is one case where you wouldn't want to use static constexper. Unless a constexpr declared object is either ODR-used or declared static, the compiler is free to not include it at all. That's pretty useful, because it allows the use of compile-time temporary constexpr arrays without polluting the compiled program with unnecessary bytes. In that case, you would clearly not want to use static, since static is likely to force the object to exist at runtime.

I honestly can't tell if this applies here or not without thinking about it more.